lstchain/reco/r0_to_dl1.py

```diff
@@ -535,9 +560,75 @@ def r0_to_dl1(
+                        # types, previous_ucts_trigger_type
+                        #
+                        if len(previous_ucts_time_unix) > 0:
+                            # keep the time & trigger type read for this
+                            # event (which really correspond to a later event):
+                            current_ucts_time = dl1_container.ucts_time
+                            current_ucts_trigger_type = dl1_container.ucts_trigger_type
+                            # put in dl1_container the proper time for this
+                            # event:
+                            dl1_container.ucts_time = \
+                                previous_ucts_time_unix.pop(0)
+                            dl1_container.ucts_trigger_type = \
+                                previous_ucts_trigger_type.pop(0)
+
+                            # now put the current values last in the list,
+                            # for later use:
+                            previous_ucts_time_unix.append(current_ucts_time)
+                            previous_ucts_trigger_type.\
+                                append(current_ucts_trigger_type)
+
+                        # Now check consistency of UCTS and Dragon times. If
+                        # UCTS time is ahead of Dragon time by more than
+                        # 1.e-6 s, most likely the UCTS info has been
+                        # lost for this event (i.e. there has been another
+                        # 'jump' of those described above), and the one we have
+                        # actually corresponds to the next event. So we put it
+                        # back first in the list, to assign it to the next
+                        # event. We also move the other elements down in the
+                        # list,  which will now be one element longer.
+                        # We leave the current event with the same time,
+                        # which will be approximately correct (depending on
+                        # event rate), and set its ucts_trigger_type to -1,
+                        # which will tell us a jump happened and hence this
+                        # event does not have proper UCTS info.
+
+                        if dl1_container.ucts_time - dl1_container.dragon_time > 1.e-6:
+                            previous_ucts_time_unix.\
+                                insert( 0, dl1_container.ucts_time)
+                            previous_ucts_trigger_type.\
+                                insert(0, dl1_container.ucts_trigger_type)
+                            dl1_container.ucts_trigger_type = -1
+
                         # Select the timestamps to be used for pointing interpolation
                         if config['timestamps_pointing'] == "ucts":
-                            event_timestamps = ucts_time_utc.unix
+                            event_timestamps = dl1_container.ucts_time
```

Comment:
are we sure this time is in the right format? I lost track after so many changes...

Reply:
I would say yes, `dl1_container.ucts_time` is defined above as `ucts_time_utc.unix` (unix format and UTC scale) and whenever it has to be corrected due to these jumps it'd always refer to `dl1_container.ucts_time` container earlier filled. One note: in OSA we use `"timestamps_pointing":"dragon"` in order to make the pointing interpolation using the corrected dragon timestamps as we agreed time ago.